prow/cmd/horologium/main.go

```diff
@@ -52,27 +52,39 @@ func main() {
 }
 
 type kubeClient interface {
-	ListJobs(map[string]string) ([]kube.Job, error)
-	CreateJob(kube.Job) (kube.Job, error)
+	ListProwJobs(map[string]string) ([]kube.ProwJob, error)
+	CreateProwJob(kube.ProwJob) (kube.ProwJob, error)
 }
 
 func sync(kc kubeClient, cfg *config.Config, now time.Time) error {
-	jobs, err := kc.ListJobs(map[string]string{"type": "periodic"})
+	jobs, err := kc.ListProwJobs(nil)
 	if err != nil {
-		return fmt.Errorf("error listing jobs: %v", err)
+		return fmt.Errorf("error listing prow jobs: %v", err)
 	}
-	latestJobs := map[string]kube.Job{}
+	latestJobs := map[string]kube.ProwJob{}
 	for _, j := range jobs {
-		name := j.Metadata.Labels["jenkins-job-name"]
+		if j.Spec.Type != kube.PeriodicJob {
+			continue
+		}
+		name := j.Spec.Job
 		if j.Status.StartTime.After(latestJobs[name].Status.StartTime) {
 			latestJobs[name] = j
 		}
 	}
 	for _, p := range cfg.Periodics {
 		j, ok := latestJobs[p.Name]
 		if !ok || (j.Complete() && now.Sub(j.Status.StartTime) > p.GetInterval()) {
-			if _, err := line.StartPeriodicJob(kc, p.Name); err != nil {
-				return fmt.Errorf("error starting periodic job: %v", err)
+			if _, err := kc.CreateProwJob(kube.ProwJob{
```

Comment:
Wonder if we need to delete ProwJob after finish, since it's a ThirdPartyResource now? Seems sinker only clean up the timedout ones.